app/routes.py

```diff
@@ -1 +1,227 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from sqlalchemy import asc, desc 
+import datetime
+import requests
+import os 
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"])
+def create_tasks():
+    request_body = request.get_json()
+    new_task = Task.from_dict(request_body) 
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    task_dict = {}
+    task_dict["task"] = {"id":new_task.task_id, "title": new_task.title,"description":new_task.description, "is_complete":bool(new_task.completed_at)}
```

Comment:
I know you mentioned that you had trouble getting this to work, but your `to_dict` implementation would have worked here!

I cloned your code and replaced this with:

```python
task_dict = {
    "task": new_task.to_dict()
}
```

And made no other changes, and it worked and passed all your existing tests. My guess the issues may come up with the addition of the `"goal_id"` key.  The way I've been suggesting writing `to_dict` to handle it looks like this, and uses an `if` block to set the `"goal_id"` key if it can:

```python
    def to_dict(self):
        new_dict = {
            "title": self.title,
            "description": self.description,
            "is_complete": bool(self.completed_at),
            "id": self.task_id,
        }

        if self.goal_id:
            new_dict["goal_id"] = self.goal_id
        
        return new_dict
```